File: backend/loaders/mongo/models/reminder.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
	"time"

	"github.com/kamva/mgm/v3"
)

type Reminder struct {
	mgm.DefaultModel `bson:",inline"`
	UserId           *primitive.ObjectID `json:"user_id" bson:"user_id"`
	Title            *string             `json:"title" bson:"title"`
	NoteId           *primitive.ObjectID `json:"note_id" bson:"note_id"`
	Description      *string             `json:"description" bson:"description"`
	RemindDate       *time.Time          `json:"remind_date" bson:"remind_date"`
}

File: client/lib/models/response/reminder/reminder.dart
class NoteReminderData {
  String id;
  String createdAt;
  String updatedAt;
  String userId;
  String title;
  String noteId;
  String description;
  String remindDate;
  bool success;

  NoteReminderData({
    required this.id,
    required this.createdAt,
    required this.updatedAt,
    required this.userId,
    required this.title,
    required this.noteId,
    required this.description,
    required this.remindDate,
    required this.success,
  });

  factory NoteReminderData.fromJson(Map<String, dynamic> json) {
    return NoteReminderData(
      id: json['id'],
      createdAt: json['created_at'],
      updatedAt: json['updated_at'],
      userId: json['user_id'],
      title: json['title'],
      noteId: json['note_id'],
      description: json['description'] ?? "",
      remindDate: json['remind_date'] ?? "",
      success: json['success'],
    );
  }
}
